### PR TITLE
add support for tls_allowed_dn_list

### PR DESCRIPTION
**Pull request by jschlyter**

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -62,6 +63,11 @@
   $sasl2_qemu_auxprop_plugin = undef,
 ) inherits ::libvirt::params {
 
+  if $tls_allowed_dn_list {
+    $tls_allowed_dn_joined =
+      join(suffix(prefix($tls_allowed_dn_list, '"'), '"'),',')
```
Comment (igalic) [inline manifests/init.pp]: how about, `join($tls_allowed_dn_list, '","')`

File: templates/libvirtd.conf.erb [hunk 2]
```diff
@@ -264,7 +264,11 @@ auth_tls = "<%= @auth_tls %>"
 # entirely rather than using empty list to disable these checks
 #
 # By default, no DN's are checked
+<%- if @tls_allowed_dn_joined -%>
+tls_allowed_dn_list = [ <%= @tls_allowed_dn_joined %> ]
```
Comment (igalic) [inline templates/libvirtd.conf.erb]: and then here we do:
```erb
tls_allowed_dn_list = [ "<%= @tls_allowed_dn_joined %>" ]
```

Comment (jschlyter) [inline templates/libvirtd.conf.erb]: That would expand the ann empty list into [ "" ], which is not really empty.

Comment (igalic) [inline templates/libvirtd.conf.erb]:
```
igalic@levix ~> puppet apply -e "fail(join(undef, '\",\"'))"
Error: Evaluation Error: Error while evaluating a Function Call, join(): Requires array to work with  at line 1:6 on node levix
```
so we're fine, cuz you have a guard if @tls_allowed_dn_joined anyway.

but i'd add an `else` branch above, which explicitly sets the variable to undef, thus, passing strict_variables = true

**Comment by jschlyter** [comment]
Are we good?

**Comment by igalic** [comment]
:+1: 
ping @thias

**Pull request closed without merging**